static/app/components/search/index.tsx

```diff
@@ -102,166 +104,147 @@ type Props = WithRouterProps<{orgId: string}> & {
-    trackAdvancedAnalyticsEvent(`${this.props.entryPoint}.open`, {
-      organization: null,
-    });
-  }
+}
 
-  handleSelect = (item: Item, state?: AutoComplete<Item>['state']) => {
-    if (!item) {
-      return;
-    }
-    trackAdvancedAnalyticsEvent(`${this.props.entryPoint}.select`, {
-      query: state && state.inputValue,
-      result_type: item.resultType,
-      source_type: item.sourceType,
+function Search(props: SearchProps): React.ReactElement {
+  React.useEffect(() => {
+    trackAdvancedAnalyticsEvent(`${props.entryPoint}.open`, {
       organization: null,
     });
+  }, [props.entryPoint]);
 
-    const {to, action, configUrl} = item;
-
-    // `action` refers to a callback function while
-    // `to` is a react-router route
-    if (action) {
-      action(item, state);
-      return;
-    }
-
-    if (!to) {
-      return;
-    }
-
-    if (to.startsWith('http')) {
-      const open = window.open();
-      if (open === null) {
-        addErrorMessage(
-          t('Unable to open search result (a popup blocker may have caused this).')
-        );
+  const handleSelectItem = React.useCallback(
+    (item: Result['item'], state?: AutoComplete<Result['item']>['state']) => {
+      if (!item) {
         return;
       }
 
-      open.opener = null;
-      open.location.href = to;
-      return;
-    }
-
-    const {params, router} = this.props;
-    const nextPath = replaceRouterParams(to, params);
-
-    navigateTo(nextPath, router, configUrl);
-  };
-
-  saveQueryMetrics = debounce(query => {
-    if (!query) {
-      return;
-    }
-    trackAdvancedAnalyticsEvent(`${this.props.entryPoint}.query`, {
-      query,
-      organization: null,
-    });
-  }, 200);
-
-  renderItem = ({resultObj, index, highlightedIndex, getItemProps}) => {
-    // resultObj is a fuse.js result object with {item, matches, score}
-    const {renderItem} = this.props;
-    const highlighted = index === highlightedIndex;
-    const {item, matches} = resultObj;
-    const key = `${item.title}-${index}`;
-    const itemProps = {...getItemProps({item, index})};
-
-    if (typeof renderItem !== 'function') {
-      throw new Error('Invalid `renderItem`');
-    }
+      trackAdvancedAnalyticsEvent(`${props.entryPoint}.select`, {
+        query: state && state.inputValue,
+        result_type: item.resultType,
+        source_type: item.sourceType,
+        organization: null,
+      });
+
+      // `action` refers to a callback function while
+      // `to` is a react-router route
+      if (typeof item.action === 'function') {
+        item.action(item, state);
+        return;
+      }
 
-    const renderedItem = renderItem({
-      item,
-      matches,
-      index,
-      highlighted,
-      itemProps,
-    });
+      if (!item.to) {
+        return;
+      }
 
-    return React.cloneElement(renderedItem, {key});
-  };
+      if (item.to.startsWith('http')) {
+        const open = window.open();
+        if (open === null) {
```

Comment:
```suggestion
        if (!open) {
```

Reply:
Do we have a styleguide for this btw? I think I'm just used to strictNullChecks being enabled (I think we should check to see if we could enable it too), so I just write it like that. I think we can just drop the negation and write this as

```
if (open) {
    open.opener = null;
    open.location.href = item.to;
    return;
} 

addErrorMessage()
```